akka-stream/src/main/scala/akka/stream/IOResult.scala

```diff
@@ -34,3 +42,12 @@ final case class IOResult private[stream] (count: Long, status: Try[Done]) {
   }
 
 }
+
+object IOResult extends AbstractFunction2[Long, Try[Done], IOResult] {
```

Comment:
`AbstractFunction2` adds the `apply`?
isn't it possible to just make the constructor public?

```scala
final case class IOResult(count: Long, status: Try[Done])
```